src/e2e/java/teammates/e2e/cases/lnp/BaseLNPTestCase.java

```diff
@@ -162,18 +163,27 @@ protected void createTestData() {
         LNPTestData testData = getTestData();
         try {
             createJsonDataFile(testData);
+            persistTestData();
             createCsvConfigDataFile(testData);
-        } catch (IOException ex) {
+        } catch (IOException | HttpRequestFailedException ex) {
             log.severe(TeammatesException.toStringWithStackTrace(ex));
         }
     }
 
     /**
      * Creates the entities in the datastore from the JSON data file.
      */
-    protected void persistTestData() {
+    protected void persistTestData() throws IOException, HttpRequestFailedException {
         DataBundle dataBundle = loadDataBundle(getJsonDataPath());
-        BackDoor.removeAndRestoreDataBundle(dataBundle);
+        String responseBody = "";
+        responseBody = BackDoor.removeAndRestoreDataBundle(dataBundle);
+
+        String pathToResultFile = createFileAndDirectory(TestProperties.LNP_TEST_DATA_FOLDER, getJsonDataPath());
+        String jsonValue = JsonUtils.parse(responseBody).getAsJsonObject().get("message").getAsString();
```

Comment:
Just a nit: The whole message body can be the json. Don't use a specific field to hold them (magic "message" field) .

Reply:
From what I checked, in order to be able to return the entire `DataBundle` object as json I need to create a new `DataBundleData` class that extends `ApiOutput`, and converts all the values into `xxxData`. I checked through the various `xxxData` classes, and my concern is that some of the classes does not have the same structure as their respective `xxxAttributes`.

For example, the class `FeedbackQuestionData` converts the two fields, `showResponsesTo` and `showGiverNameTo`, which are `FeedbackParticipantType` class, into `FeedbackVisibilityType` class. I'm worried that using `xxxData` will have unintended side effect - should we proceed with this? (Removing the magic "message" field)

Reply:
I am not suggestion to use `xxxData`. Let me rephrase the question, why there is `message` key here?

Reply:
This is a behaviour from the `PutDataBundleAction.java`, method execute(). 
[Link to code](https://github.com/TEAMMATES/teammates/blob/c2293eec588f0a11399ec8c02ec8cce57318f376/src/main/java/teammates/ui/webapi/action/PutDataBundleAction.java#L27-L41)

The execute() method has to return a `JsonResult` object. `JsonResult` class has 4 constructors : 
```
public JsonResult(ApiOutput output) { }
public JsonResult(ApiOutput output, List<Cookie> cookies) { }
public JsonResult(String message) { }
public JsonResult(String message, int statusCode) {}
```
As for now I'm returning the String converted, it will use the 3rd constructor which creates a `MessageOutput` object with my json String as the message. Which is why the `message` key is used.

In order to use the `ApiOutput` I need to create the `DataBundleData` class, but now that I think about it the message body will still be returned under the "DataBundle" key even if I do that.

I've tried looking around to find a way to remove the extra key (basically pass the entire json string by itself), but I couldn't find it so far. Not sure whether creating an entire group of extra classes is worth it either.